File: src/Backend/Inspections.API/Features/Checklists/Models/CheckListItemDTO.cs
using Inspections.Core.Domain.CheckListAggregate;

namespace Inspections.API.Features.Checklists.Models;

public class CheckListItemDTO
{
    public int Id { get; set; }
    public int CheckListId { get; set; }
    public string Text { get; set; } = default!;
    public CheckValue Checked { get; set; }
    public bool Editable { get; set; }
    public bool Required { get; set; }
    public string? Remarks { get; set; }
}

File: src/ClientApp/types/CheckLists/ViewModels/CheckListItemDTO.ts
import { CheckValue } from '../Models/CheckValue'
import { CheckListParamDTO } from './CheckListParamDTO'

export interface CheckListItemDTO {
  id: number
  checkListId: number
  text: string
  checked: CheckValue
  required: boolean
  remarks: string
  textParams: CheckListParamDTO[]
}
